api/controllers/pet.go

```diff
@@ -82,3 +82,23 @@ func (cntrl *PetController) ListUserPets(w http.ResponseWriter, r *http.Request)
 
 	w.WriteHeader(http.StatusOK)
 }
+
+func (cntrl *PetController) ListByUserNoAuth(w http.ResponseWriter, r *http.Request) {
+    pets, err := cntrl.Usecase.ListByUserNoAuth()
+    if err != nil {
+        http.Error(w, err.Error(), http.StatusInternalServerError)
+        return
+    }
+
+	if len(pets) == 0 {
+		w.WriteHeader(http.StatusNoContent)
```

Comment:
Acredito que o status code ideal caso o retorno seja um lista vazia seria o 200. 

O status 204 é idealmente usado para endpoints que, em todas das chamadas, não retornarão conteúdo (DELETE, por exemplo). Como no caso desse endpoint se trata de uma consulta ao recurso de pets e o usuário sempre esperará um body na resposta da API, acredito que o ideal seja retorna um 200 com o array vazio.

A primeira resposta dessa thread tem uma boa explicação [Link](https://stackoverflow.com/questions/13366730/proper-rest-response-for-empty-table)

Reply:
@devlefel, mas se você aprovar, por mim de boa.